safety_limiter/src/safety_limiter.cpp

```diff
@@ -576,9 +578,12 @@ class SafetyLimiterNode
   limitMaxVelocities(const geometry_msgs::Twist& in)
   {
     auto out = in;
-    out.linear.x = (out.linear.x > 0) ?
-                       std::min(out.linear.x, max_values_[0]) :
-                       std::max(out.linear.x, -max_values_[0]);
+    const double vel_ratio = max_values_[0] / std::hypot(out.linear.x, out.linear.y);
```

Comment:
Is it safe when `out.linear = { x: 0, y: 0 }`? (divide by zero)

Reply:
Fixed